test_runner/src/main/kotlin/ftl/shard/Shard.kt

```diff
@@ -1,123 +1,60 @@
 package ftl.shard
 
-import ftl.reports.xml.model.JUnitTestSuite
-import ftl.util.Utils.fatalError
+import ftl.reports.xml.model.JUnitTestResult
 
 data class TestMethod(
     val name: String,
     val time: Double
 )
 
 data class TestShard(
-    val time: Double,
+    var time: Double,
     val testMethods: MutableList<TestMethod>
 )
 
-data class TestShards(
-    val shards: MutableList<TestShard> = mutableListOf()
-)
-
 object Shard {
 
     // take in the XML with timing info then return list of shards
-    private fun calculateShardsByTime(testSuite: JUnitTestSuite, maxTestShards: Int) {
-
-        // read in test methods + time from XML
-        // calculate shard time
-
-        // if maxTestShards >= # of tests ... -1 sharding. 1 test per shard.
-        // tests > maxTestShards
+    fun calculateShardsByTime(runningTests: List<String>, testResult: JUnitTestResult, maxShards: Int): List<TestShard> {
 
+        val junitMap = mutableMapOf<String, Double>()
 
-        // TODO: test / rewrite this.
-        testList.sortByDescending { it.time }
-
-        val shardList = TestShards()
-        var shardCount = 0
-        var startingDuration = testList.first().time
-
-        while (testList.size > 0) {
-            if (shardCount == maxTestShards) {
-                // we have "filled" all shards, reset starting duration/position
-                startingDuration = testList.first().time
-                System.out.println("Starting duration $startingDuration")
-                shardCount = 0
+        // Create a map with information from previous junit run
+        testResult.testsuites?.forEach { testsuite ->
+            testsuite.testcases?.forEach { testcase ->
+                junitMap[testcase.name] = testcase.time.toDouble()
```

Comment:
testcase name isn't unique. The fully qualified name is `suite.name + testCase.classname + testCase.name`

https://github.com/TestArmada/flank/blob/bd8dd925f6f10a61fe583b03b63228daa0f6a885/test_runner/src/main/kotlin/ftl/reports/HtmlErrorReport.kt#L32

Reply:
great! I wanted to test that by hooking with a real run.